File: samples/r/theme.R
# Functions, named lists, data frames, formulas, pipes, and vectorization.
theme <- list(
  name = "CodePen Theme Original",
  accent = "#96b38a",
  tokens = c("keyword", "string", "comment")
)

label_theme <- function(value, prefix = "theme") {
  stopifnot(is.list(value), nzchar(value$name))
  sprintf("%s: %s (%s)", prefix, value$name, value$accent)
}

visible_tokens <- function(tokens, include_comments = FALSE) {
  selected <- if (include_comments) tokens else tokens[tokens != "comment"]
  toupper(selected)
}

token_table <- data.frame(
  role = theme$tokens,
  color = c("#ddca7e", "#96b38a", "#717790"),
  visible = c(TRUE, TRUE, FALSE)
)

score <- with(token_table, as.integer(visible) ~ role)
result <- token_table |>
  subset(visible) |>
  transform(label = paste(role, color, sep = ": "))

message(label_theme(theme))
print(result[, c("role", "label")])

File: samples/julia/theme.jl
module CodePenTheme

export Theme, label, visible_tokens

# Structs, typed fields, keyword arguments, comprehensions, and macros.
struct Theme{T<:AbstractString}
    name::T
    accent::T
    tokens::Vector{Symbol}
end

function label(theme::Theme; prefix::AbstractString = "theme")
    "$(prefix): $(theme.name) ($(theme.accent))"
end

visible_tokens(theme::Theme) = [
    uppercase(String(token))
    for token in theme.tokens
    if token !== :comment
]

function contrast_score(background::AbstractString, accent::AbstractString)::Int
    return abs(length(background) - length(accent)) + 7
end

const ORIGINAL = Theme(
    "CodePen Theme Original",
    "#96b38a",
    [:keyword, :string, :comment],
)

@info label(ORIGINAL) tokens = visible_tokens(ORIGINAL)
println(contrast_score("#1d1e22", ORIGINAL.accent))

end
